src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -67,10 +67,14 @@
      (handler (assoc request :jruby-instance jruby-instance)))))
 
 (defn get-environment-from-request
-  "Gets the environment from a request."
+  "Gets the environment from the URL or query string of a request."
   [req]
-  (-> req
-      (get-in [:params "environment"])))
+  ;; :environment is the value of a path component, "environment" is the value
+  ;; of a query string parameter. If environment is an inherent part of the
+  ;; resource being requested (part of the path), we prefer that so it doesn't
+  ;; get overridden by a rogue query string parameter.
+  (or (get-in req [:params :environment])
+      (get-in req [:params "environment"])))
```

Comment:
@cprice404 Yeah, shorter-term fix which allows this to get merged would be fine.  I'm fine with just changing the first line to look for the param in `route-params` else fall back to the `params` map.  The `params` map would have the one from the query-string, if present, else the one from a urlencoded-form parameter, if present, because it will have been culled from the Ring middleware into the `params` map by this point.